### ISSUE TITLE
Colour

### ISSUE THREAD

**Issue by govuk-design-system** [alphagov/govuk-design-system-backlog#38]
Use this issue to discuss [this style in the GOV.UK Design System](https://govuk-design-system-production.cloudapps.digital/styles/colour/).

**Comment by owenm6** [comment]
Looks like the colour page is missing the button colour. Can create an issue, but didn't know whether this was a deliberate choice or mistake.

**Comment by nickcolley** [comment]
Talking about which colours contrast well with each other:

jamesbuller_ho [12:42 PM]
> It would be nice if they listed the acceptable combinations to save you and everyone else testing them yourselves

**Comment by accessiblewebuk** [comment]
Assuming there aren't too many colours in the overall palette, it would be best to show all possible allowable combinations

**Comment by Nooshu** [comment]
After using the colour palette I'm wondering if all the colours should be mapped to a variable name? I was having to jump back and forth between code and documentation to see if a particular colour was already a predefined variable or a reference in the `govuk-colour` map. e.g.

```
$govuk-colour-black;
$govuk-colour-grey-1;
$govuk-colour-grey-2;
$govuk-colour-grey-3;
$govuk-colour-grey-4;
$govuk-colour-white;
```

**Comment by joelanman** [comment]
alternatively use the function for everything, either way you don't have to think about which method to use for which colour:

```
govuk-colour('error')
govuk-colour('focus')
govuk-colour('black')
```

I wonder if the variable name approach is just more expected to our users.

**Comment by Nooshu** [comment]
Yes that could work too. Personally I prefer the variable as it looks less "clunky" in lines like this:

```
background: transparentize(govuk-colour('grey-4'), 0.05);

background: transparentize(govuk-colour-grey-4, 0.05);
```

**Comment by 36degrees** [comment]
> After using the colour palette I'm wondering if all the colours should be mapped to a variable name?

As this is an implementation detail, would you mind moving this discussion to an issue on the Frontend repo?

**Comment by Nooshu** [comment]
Further colour implementation discussion moved to the Frontend repo [here](https://github.com/alphagov/govuk-frontend/issues/920).

**Comment by abbott567** [comment]
There is an issue with $grey-4 (f8f8f8) in that it doesn't show up at all under compression. So all of our DWP equipment that our staff use essentially stream a desktop from a server somewhere and grey-4 just shows as white. The issue came to light when testing the tab component as the background colour of the tabs is grey-4 and doesn't show up at all. I have raised this issue in #100 also. Is there a need for grey-4 at all? Unlike the govuk-blue and link-colour etc there seems to be no semantic use for grey-4. So I wonder if it needs a warning or to be removed so people don't add it to their design and unknowingly introduce usability issues.

**Comment by tribute3108** [comment]
#a1acb2 - used for footer top border - doesn't seem to be defined anywhere as a variable. Can anyone enlighten me please?

**Comment by fofr** [comment]
Like @owenm6, I'm wondering why the button green is missing from the colours list.

`#00823b` is missing. There is only a dark green, `#006435`.

**Comment by alaricking** [comment]
We had an accessibility review on a service we launched at Crown Commercial Service, where they flagged and failed the contrast between the focus colour (yellow) #FFBF47 and background colour (white) #FFFFFF; citing the ratio should be 3:1 (currently its 1.6:1). 

Worth noting that there were no strong complaints from users during testing though.

**Comment by nickcolley** [comment]
@alaricking thanks for letting us know, we're doing some work on that very soon: https://github.com/alphagov/govuk-frontend/issues/1137

**Comment by benchilds** [comment]
I think the link to the Organisation colour file needs updating? The Colours page at https://design-system.service.gov.uk/styles/colour/ has a link to https://github.com/alphagov/govuk-frontend/blob/master/src/settings/_colours-organisations.scss that throws a 404. It looks like the link should be https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/settings/_colours-organisations.scss.

* Old: https://github.com/alphagov/govuk-frontend/blob/master/src/settings/_colours-organisations.scss
* New: https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/settings/_colours-organisations.scss

**Comment by 36degrees** [comment]
Thanks for pointing that out, @benchilds – I've [raised an issue](https://github.com/alphagov/govuk-design-system/issues/934) for it.

**Comment by 36degrees** [comment]
@benchilds the broken link has now been fixed – thanks for pointing that out.

**Comment by peteryates** [comment]
We've had the results back from our DAC (Digital Accessibility Centre) report and there was a problem raised by them regarding our focus colour choice.

To quote the report directly:

> # Medium priority
> Low colour contrast ratio (focus indicator)
> The focus indicator fails colour contrast
> ## WCAG Reference:
> #### Success Criterion 1.4.11 Non-text Contrast (Level AA)
> #### Understanding Non-text Contrast | How to Meet Non-text Contrast
> Issue ID: DAC- Low colour contrast ratio (focus indicator)
> Screen Shot:
>
> ![Screenshot 2019-07-23 at 12 36 55](https://user-images.githubusercontent.com/128088/61709214-94f65b80-ad46-11e9-9b7e-17c0954fa74b.png)
>
> The current visible focus indicator colour does not meet the required ratio of 3:1 for AA standard. This may be problematic for visually impaired users. The current ratio is 1.64:1.
This is consistent throughout the service where this colour has been used as the focus indicator.
>
> #### Code ref(s)
> Colour contrast analyst comments:
> “The color combination found for the link highlighting mechanism did not pass the ratio of 3:1 against adjacent colours.”
> #### Solution
> Ensure the colour contrast ratio for adjacent colour meet the minimum required ratio of 3:1.

Has this been highlighted by anyone else or any other DAC-style tests? I had a look but couldn't find any references to focus colour specifically. A slightly-darker shade of amber might suffice? 🤷🏽‍♂️

**Comment by amyhupe** [comment]
Hi @peteryates

Thanks for your comment. I work in the GOV.UK Design System team 

We are aware of this issue and are currently making a [number of changes](https://design-system.service.gov.uk/accessibility/)
to the Design System and Frontend to make sure it's WCAG 2.1 AA compliant. We'll be releasing an update containing these changes within the next week.

If you can hold fire until then, the new version should fix your issues. If you need any help with updating, you can get in touch with the team on #govuk-design-system on Slack or by emailing govuk-design-system-support@digital.cabinet-office.gov.uk

I hope that helps.

Amy

**Comment by emma-cuthbert** [comment]
Yesterday I observed a user research session with a member of DWP staff who has dyslexia and problems with short-term memory. She said she finds it very difficult to read text on a white background (her words: "it jumbles everything up for me") and much prefers grey. She specifically mentioned the grey background we use in the footer of gov.uk pages as being much clearer for her.

**Comment by tomyems** [comment]
The Home Office Design System, which is used for internal products, uses a grey background for this reason. (Source: https://design.homeoffice.gov.uk/styles/colour)

**Comment by jlarmstrongiv** [comment]
Would it be possible to add tints and shades like the [USWDS](https://designsystem.digital.gov/design-tokens/color/overview/)? 

![image](https://user-images.githubusercontent.com/20903247/161348566-72af0957-a58c-46c7-bb3c-ed123974be8b.png)

**Comment by joelanman** [comment]
@jlarmstrongiv sorry for the delay, it would be good to know about what context users need it in - can you share any details?

**Comment by jlarmstrongiv** [comment]
@joelanman sure, here is a previous discussion on the topic https://github.com/uswds/uswds/issues/2255 I’m curious to know your thoughts

**Comment by querkmachine** [comment]
Having lighter and darker shades of our colour palette will probably be necessary if and when we get around to tackling #177, though whether we provide a full gamut of colours or only tweak the ones that need tweaking is not something that's been discussed yet. 

Relatedly, [we already use darkened shades of departmental colours](https://github.com/alphagov/govuk-frontend/blob/3c30a141ff69227a2111d235f9d53f029e61cdb4/src/govuk/settings/_colours-organisations.scss) when we need to provide sufficient contrast for text on white.

**Comment by dkryaklin** [comment]
Hey there! I am working on my a11y colour tool. And checking some open source public colour palettes for a11y issues. I've prepared this visual to show possible improvements. Let me know what you think. For me personally feels like improvement. Happy to contribute more.

<img width="1800" height="1520" alt="Image" src="https://github.com/user-attachments/assets/2881de78-77f0-4b70-bdac-206f3218d359" />

| Colour | Now | Proposed | On white now | On white proposed | Why |
| --- | --- | --- | --- | --- | --- |
| brown | `#99704a` | `#8e6640` | 4.40:1, fails AA | 5.09:1, passes AA | Same hue. White on it goes from 4.40:1 to 5.09:1 too. Brown is the only group with no shade. |
| error / red | `#ca3535` | `#c73233` | 5.15:1, APCA Lc 74.1 | 5.34:1, APCA Lc 75.1 | Passes AA already. This only clears APCA 75 for body text. Optional. |

Also worth a note on the colour page: which palette colours pass AA as body text on white. Orange, yellow, brown and teal accent do not. Every tint-25 except purple does not. Every shade does, except yellow shade-25.
